app/conf/config.go

```diff
@@ -79,8 +83,144 @@ func LoadYamlConfig(filename string, env string) []string {
 		app.CurrentEnvironment.Port = "4100"
 	}
 
-	app.SyncQueues.Lock()
+	err = createSqsQueues(env)
+	if err != nil {
+		return ports
+	}
+
+	err = createSNSTopics(env)
+	if err != nil {
+		return ports
+	}
+
+	return ports
+}
+
+func StartWatcher(filename string, env string) {
+	quit := make(chan struct{})
+	//create watcher
+	if filename == "" {
+		filename, _ = filepath.Abs("./app/conf/goaws.yaml")
+	}
+	log.Infof("Starting watcher on file: %v", filename)
+
+	watcher, err := fsnotify.NewWatcher()
+	defer watcher.Close()
+
+	if err != nil {
+		log.Errorf("err: %s", err)
+	}
+
+	// Start listening for events.
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				if event.Has(fsnotify.Remove) {
```

Comment:
After looking at this a bit deeper, I don't think I fully understand this block.  Is it because in your environment - this "remove" event is what happens on a file change (in a "remove" then "recreate" pattern)?  If so, the file name isn't changing is it?  Could we not just do the wait/loop to check if it's rebuilt and then proceed down with the rest of the reloading calls?

I think I'm confused about 2 things really.
1. Why call `StartWatcher` again, and synchronously?
2. Why block the outer function with the `quit` channel?  In most of these non-k8 with the remove event cases would that ever actually hang up or is that goroutine just orphaned now?